modules/library/main/src/main/java/org/geotools/geometry/jts/ReferencedEnvelope.java

```diff
@@ -822,8 +968,25 @@ public static ReferencedEnvelope create(
      * @return ReferencedEnvelope, ReferencedEnvelope3D if it is 3d
      */
     public static ReferencedEnvelope create(ReferencedEnvelope env, CoordinateReferenceSystem crs) {
-        return create((org.geotools.api.geometry.Envelope) env, crs);
+        return create((Bounds) env, crs);
+    }
+
+    /**
+     * Construct 2D WGS84 referenced envelope using rectangle conventions using width and height.
+     *
+     * <p>Migration method, previously {@code Envelope2D(crs,x,y,w,h)}.
+     *
+     * @param x
+     * @param y
+     * @param width
+     * @param height
+     * @return ReferencedEnvelope WGS84
+     */
+    public static ReferencedEnvelope envelope2D(
+            CoordinateReferenceSystem crs, double x, double y, double width, double height) {
+        return new ReferencedEnvelope(x, x + width, y, y + height, crs);
     }
```

Comment:
Wondering, should this method be inlined and removed, or made into a constructor leveraging the different order of arguments? Seeing "ReferencedEnvelope.evenvelope2D" around the codebase is weird... and the name is long too.  Thinking out loud, if you want to keep the static method, what about "ReferencedEnvelope.rect(x,y,w,h,crs)`` since it's using the rectangle convention?

Reply:
ReferencedEnvelope has both constructors (which are easy to misuse combing 2D data with 3D CRS) and factory methods which create ReferencedEnvelope or ReferencedEnvelope3D based on CRS provided. 

I will rename to `rect(crs,x,y,w,h)` - the order was provided so there is an "easy" search and replace to perform when migrating from Envelope2D. I am happy to deprecate the method, or if it is not useful at the end of the day it can be removed.